ivy/functional/frontends/paddle/nn/functional/norm.py

```diff
@@ -8,3 +8,34 @@
 @with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
 def layer_norm(x, normalized_shape, weight=None, bias=None, epsilon=1e-05, name=None):
     return ivy.layer_norm(x, normalized_shape, weight, bias, epsilon)
+
+
+# batch_norm
+@to_ivy_arrays_and_back
+@with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
```

Comment:
Out of curiosity, how did you arrive at these `supported_dtypes`, did you create a simple script to test the supported dtypes?

Reply:
You can modify the following script to generate the values for the parameters of batch_norm and then call `paddle.bacth_norm` instead of `paddle.numel`:

```python
#, "uint16", "bfloat16"
dtypes_list = ["int8", "int16", "int32", "int64", "uint8", "float16", "float32", "float64", "complex64", "complex128", "bool",]
unsupported = []
supported = []
for dtype in dtypes_list:
    try:
        a = paddle.to_tensor([[[9.]]],dtype=dtype)
        # insert the function you want to test in the following line
        #paddle.normal(a,a)
        paddle.numel(a)
        supported.append(a.dtype.name)
    except:
        # use this to generate native dtype
        unsupported.append(a.dtype.name)
        # use this to generate string dtype
        #unsupported.append(dtype)
print(unsupported)
print(supported)
```

Reply:
Oh, ok I will modify it. 

I also wrote this function, batch_norm_data to generate data for batch_norm, that I wanted you to check:

# global
from hypothesis import strategies as st

# local
import ivy_tests.test_ivy.helpers as helpers
from ivy_tests.test_ivy.helpers.testing_helpers import handle_frontend_test

from ivy_tests.test_ivy.test_functional.test_nn.test_norms import (
    _generate_data_layer_norm,
)
@st.composite
def batch_norm_data(draw):
    # Define the strategies for each parameter of the batch_norm function
    x_strategy = draw(st.lists(st.floats()))  
    mean_strategy = draw(st.lists(st.floats()))
    var_strategy = draw(st.lists(st.floats())) 
    offset_strategy = draw(st.lists(st.floats())) 
    scale_strategy = draw(st.lists(st.floats()))
    training_strategy = draw(st.floats()) 
    epsilon_strategy = draw(st.floats()) 
    momentum_strategy = draw(st.floats()) 
    data_format_strategy = draw(st.text()) 
    out_strategy = draw(st.lists(st.floats()))

    # Return a tuple of the drawn values
    return x_strategy, mean_strategy, var_strategy, offset_strategy,scale_strategy, training_strategy,epsilon_strategy,momentum_strategy, data_format_strategy, out_strategy 


I don't know if it looks all right. But also I am getting some error running it.

Reply:
Hi @hello-fri-end, just an update: I worked on it as instructed. But I am running on some errors due to machine (Unable Windows Long Path). I troubleshoot it and get back to you soon.